packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIteratorItem.tsx

```diff
@@ -94,24 +96,9 @@ export const SimpleFormIteratorItem = React.forwardRef(
                             inline && SimpleFormIteratorClasses.inline
                         )}
                     >
-                        {Children.map(
-                            children,
-                            (input: ReactElement, index2) => {
-                                if (!isValidElement<any>(input)) {
-                                    return null;
-                                }
-                                const { source, ...inputProps } = input.props;
-                                return cloneElement(input, {
-                                    source: source
-                                        ? `${member}.${source}`
-                                        : member,
-                                    index: source ? undefined : index2,
-                                    resource,
-                                    disabled,
-                                    ...inputProps,
-                                });
-                            }
-                        )}
+                        <SourceContextProvider value={sourceContext}>
+                            {children}
+                        </SourceContextProvider>
```

Comment:
Why do you add the SourceContext in each SimpleFormIteratorItem instead of once in the wrapper SimpleFormIterator?

Reply:
Because `member` actually contains the index